w3af/plugins/audit/xss.py

```diff
@@ -150,19 +156,15 @@ def _identify_trivial_xss(self, mutant):
         
         response = self._uri_opener.send_mutant(trivial_mutant)
 
+        # special character caused a json response,
+        # return None to close the later search
+        if self._is_json_response(response):
+            return None
```

Comment:
If the code gets to this point, and `self._check_persistent_xss` is True, then we're not running:

```
self._xss_mutants.append((trivial_mutant, response.id))
```

I wouldn't change the order of these calls, just in case.